ambry-account/src/main/java/com/github/ambry/account/CachedAccountService.java

```diff
@@ -632,6 +729,28 @@ private Pair<List<Container>, List<Container>> getUpdatedContainers(Account acco
     return new Pair<>(addedContainers, updatedContainers);
   }
 
+  /**
+   * A helper function to get account and container id from its name.
+   * @param accountName the name of the account.
+   * @param containerName the name of the container.
+   * @return the pair of accountId and containerId.
+   * @throws AccountServiceException
+   */
+  private Pair<Short, Short> getAccountAndContainerIdFromName(String accountName, String containerName)
+      throws AccountServiceException {
+    final Container container = getContainerByName(accountName, containerName);
+    if (container == null) {
+      throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
+          AccountServiceErrorCode.BadRequest);
+    }
+    final short accountId = container.getParentAccountId();
+    final short containerId = container.getId();
+    if (mySqlAccountStore == null) {
+      mySqlAccountStore = this.supplier.get();
```

Comment:
Minor: Just thinking that getting mysqlAccountStore here is little out of place. May be we can have within addDataset, getDataset, updateDataset itself although it leads to duplicated code?

Reply:
sure, updated.

Reply:
Not sure why I could not leave comments for below one, so I just reply here. yes, I'll disable config.ignoreNewDatabaseUploadError before setting config.enableGetFromNewDbOnly. so the step would be:
1. support dual upload
2. one time dump data
3. once it's in sync and don't see any issue, disable config.ignoreNewDatabaseUploadError
4. after a while when it's stable, enable config.enableGetFromNewDbOnly to get from new db only
5. disable upload to old db.